Source/WebCore/css/CSSGroupingRule.cpp

```diff
@@ -110,10 +111,51 @@ ExceptionOr<void> CSSGroupingRule::deleteRule(unsigned index)
     return { };
 }
 
-void CSSGroupingRule::appendCSSTextForItems(StringBuilder& result) const
+String CSSGroupingRule::cssText() const
 {
-    for (unsigned i = 0, size = length(); i < size; ++i)
-        result.append("  ", item(i)->cssText(), '\n');
+    StringBuilder builder;
+
+    builder.append(" {");
+
+    auto [decls, rules] = cssTextForItems();
+
+    if (decls.isEmpty() && rules.isEmpty()) {
+        builder.append("\n}");
+        return builder.toString();
+    }
+
+    if (rules.isEmpty()) {
+        builder.append(' ');
+        builder.append(decls);
+        builder.append(" }");
+        return builder.toString();
+    }
+    
+    if (decls.isEmpty()) {
+        builder.append(rules);
+        builder.append("\n}");
+        return builder.toString();
+    }
+
+    builder.append('\n');
+    builder.append(decls);
+    builder.append(rules);
+    builder.append("\n}");
```

Comment:
You can give multiple parameters to append and do this in a single line.

Reply:
I think there is no overload of append() which takes a mix of StringBuilder and String. Maybe it's worth adding one though ?

Reply:
There is an overload with StringBuilder&& though which is more efficient

Reply:
Actually no it's a FIXME to support StringBuilder : https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/text/StringBuilder.h#L70